solutions/src/main/java/ru/sberbank/school/task10/ScalableThreadPool.java

```
@@ -0,0 +1,144 @@
+        }
+        stopped = true;
+    }
+
+    @Override
+    public void execute(Runnable runnable) {
+        synchronized (threads) {
+            if (threads.isEmpty()) {
+                throw new NoSuchElementException("No threads running! Call start() first");
+            }
+        }
+        synchronized (tasks) {
+
+            tasks.add(runnable);
+
+            synchronized (threadID) {
+                if ((threads.size() < max) && (threadID.size() == 0)) {
+                    if (count < max) {
+                        ++count;
+                        threadID.add(count);
+                    }
+                    newThread().start();
+                }
+            }
+
+            tasks.notify();
+
+        }
+
+    }
+
+    private Thread newThread() {
+
+
+        int id = threadID.poll();
+
+        Thread thread = new Thread("Thread-" + id) {
+
+            @Override
+            public void run() {
+                Runnable runnable;
+                while (!Thread.interrupted()) {
+                    synchronized (tasks) {
+                        if (tasks.isEmpty()) {
+                            try {
+                                threadID.add(id);
+                                if (threads.size() > min) {
+                                    threads.remove(Thread.currentThread());
+                                    Thread.currentThread().interrupt();
+                                } else {
+                                    tasks.wait();
+                                }
+
+                            } catch (InterruptedException e) {
+                                System.out.println();
+                                Thread.currentThread().interrupt();
+                            }
+                        }
+                        runnable = tasks.poll();
+                    }
+                    try {
+                        runnable.run();
+                    } catch (NullPointerException ignored) {
```

Empty catch block.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a843155985>*